internal/gps/checks.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import (
+	"github.com/golang/dep/internal/gps/pkgtree"
+)
+
+func FindIneffectualConstraints(manifest Manifest, packageTree pkgtree.PackageTree) []ProjectRoot {
+	var stdLibFn func(string) bool
+	// will track return value here
+	ineffectuals := make([]ProjectRoot, 0, 10)
```

Comment:
i suspect we're going to need a more complex type, here. at minimum, we need to differentiate between ineffectual ignores and ineffectual constraints.